### PR TITLE
Adding Timeout to e2e http.Client settings

### PR DESCRIPTION
**Pull request by pkedy**
# Description

Per https://medium.com/@nate510/don-t-use-go-s-default-http-client-4804cb19f779, adding `Timeout` to the e2e http.Client initialization.

Lets see how the e2e go 🤞

### CODE REVIEW

File: tests/e2e/utils/helpers.go [hunk 1]
```diff
@@ -78,6 +78,7 @@ func GenerateRandomStringKeyValues(num int) []SimpleKeyValue {
 func newHTTPClient() http.Client {
 	doOnce.Do(func() {
 		defaultClient = http.Client{
+			Timeout: time.Second * 15,
```
Comment (artursouza) [inline tests/e2e/utils/helpers.go:81]: We have seen some calls taking over 1 minute on Windows. Shouldn't this timeout match the one below?

Comment (pkedy) [reply tests/e2e/utils/helpers.go:81]: I wanted to start with a sane value to see if it the timeouts are Dial/Connection or Read timeouts. Tests on this iteration passed so maybe the issues have been Dial/Connection related. I think we should re-trigger the tests a few times though to see what happens.

**Comment by wcs1only** [comment]
E2E failure is a known unrelated issue #3031

**Review by wcs1only — APPROVED** [review]
A 15s read timeout is reasonable, and seems to work ok, now that we fixed the issue that was causing TCP connections to hang indefinitely on Windows.

File: tests/e2e/utils/helpers.go (hunk 1 above)
Comment (wcs1only) [reply tests/e2e/utils/helpers.go:81]: So the issue we were having with windows networking was causing reads() to hang forever, hence why early runs of this were hitting context timeouts. Now that we've fixed this, we shouldn't expect reads to exceed 15s. This is a sane value.

**Review by artursouza — APPROVED** [review]

**Pull request merged**